airbyte-api/src/main/openapi/config.yaml

```diff
@@ -2177,6 +2457,65 @@ components:
           type: array
           items:
             $ref: "#/components/schemas/SourceDefinitionRead"
+    CustomSourceDefinitionCreate:
+      type: object
+      required:
+        - workspaceId
+        - sourceDefinition
+      properties:
+        workspaceId:
+          $ref: "#/components/schemas/WorkspaceId"
+        sourceDefinition:
+          $ref: "#/components/schemas/SourceDefinitionCreate"
+    CustomSourceDefinitionUpdate:
+      type: object
+      required:
+        - workspaceId
+        - sourceDefinitionUpdate
+      properties:
+        workspaceId:
+          $ref: "#/components/schemas/WorkspaceId"
+        sourceDefinition:
+          $ref: "#/components/schemas/SourceDefinitionUpdate"
+    CustomSourceDefinitionDelete:
+      type: object
+      required:
+        - sourceDefinitionId
+        - workspaceId
+      properties:
+        sourceDefinitionId:
+          $ref: "#/components/schemas/SourceDefinitionId"
+        workspaceId:
+          $ref: "#/components/schemas/WorkspaceId"
+    PrivateSourceDefinitionRead:
+      type: object
+      required:
+        - sourceDefinition
+        - granted
+      properties:
+        sourceDefinition:
+          $ref: "#/components/schemas/SourceDefinitionRead"
+        granted:
+          type: boolean
+    PrivateSourceDefinitionReadList:
+      type: object
+      required:
+        - sourceDefinitionGrants
+      properties:
+        sourceDefinitionGrants:
+          type: array
+          items:
+            $ref: "#/components/schemas/PrivateSourceDefinitionRead"
+    SourceDefinitionGrantUpdate:
+      type: object
+      required:
+        - sourceDefinitionId
+        - workspaceId
+      properties:
+        sourceDefinitionId:
+          $ref: "#/components/schemas/SourceDefinitionId"
+        workspaceId:
+          $ref: "#/components/schemas/WorkspaceId"
```

Comment:
This happens to be the same schema as `CustomSourceDefinitionDelete`, should we consolidate them?

Reply:
I think it's fine to leave separate, DRY wouldn't be getting us a ton of value here I don't think

Reply:
Yeah, that was my original thinking so I kept them separate.

However I just realized that I should also add a get route also with workspaceId (we don't want one workspace to be able to get the details of a custom definition that doesn't belong to it).

At that point, it seems like there are 4 endpoints that require this combo of actorId/workspaceId, so I ended up consolidating them, will push that change and you can see whether it looks reasonable.

Reply:
Yep after looking at the change I think consolidation is definitely the right call here